library/src/blas1/rocblas_iamaxmin_template.h

```diff
@@ -0,0 +1,225 @@
+#pragma once
+
+#include "rocblas_batched_arrays.h"
+
+#ifndef AMAX_AMIN_REDUCTION
+#error undefined macro AMAX_AMIN_REDUCTION
+#endif
+
+#define QUOTE2(S) #S
+#define QUOTE(S) QUOTE2(S)
+
+#define JOIN2(A, B) A##B
+#define JOIN(A, B) JOIN2(A, B)
+
+//
+//
+// Template on the reduction types.
+//
+//
+template <typename T>
+struct reduction_types
+{
+public:
+    using Ti = T;
+    using To = T;
+};
+
+//
+// Specialization of the reduction types for the float complex.
+//
+template <>
+struct reduction_types<rocblas_float_complex>
+{
+public:
+    using Ti = rocblas_float_complex;
+
+public:
+    using To = float;
+};
+
+//
+// Specialization of the reduction types for the double complex.
+//
+template <>
+struct reduction_types<rocblas_double_complex>
+{
+public:
+    using Ti = rocblas_double_complex;
+
+public:
+    using To = double;
+};
+
+//
+// Extension of a rocblas_int as a pair of index and value.
+// As an extension the index must be stored first.
+// This struct is the working type, i.e. the intermediate data type of the min/max routine.
+//
+template <typename T>
+struct index_value_t
+{
+    rocblas_int index;
+    T           value;
+};
+
+//
+// Overload the output stream operator for the intermediate data type.
+//
+template <typename T>
+std::ostream& operator<<(std::ostream& out, const index_value_t<T>& index_value)
+{
```

Comment:
`index_value` is an internal implementation type. Why do we need IO overloads for it?

Reply:
Debugging convenience

Reply:
> Debugging convenience

Please `#if 0` or comment it out